sdk/rust-language-support/examples/image-processing/src/main.rs

```diff
@@ -0,0 +1,51 @@
+//! A simple image processing example
+//!
+//! ## Context
+//!
+//! A relatively-low powered device (think e.g. an M-class microcontroller, or similar)
+//! wants to offload some heavy, complex calculation to a third-party---image processing here.
+//! The owner of the device wishes to ensure that the computation was faithfully executed---that is,
+//! the results are trustworthy and are definitely the output of the image transform---and also are
+//! kept secret from third parties wishing to see what is happening on the device.
+//!
+//! Inputs:                  1.
+//! Assumed form of inputs:  A PNG image mapped to the virtual filesystem at runtime
+//! Ensured form of outputs: A PNG image mapped saved to the virtual filesystem
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+extern crate image;
+
+use image::{GenericImageView, imageops, ImageFormat};
+use std::process::exit;
+use wasi_types::ErrNo;
+
+fn process_image() -> Result<(), wasi_types::ErrNo> {
```

Comment:
Can you add top-level `rustdoc` comments to this function, explaining what it does?  I think the comment about `jpeg` decoding and Rayon threads can be moved into the `rustdoc` comment, too.